packages/core/src/lib/TwaManifest.ts

```diff
@@ -315,6 +315,139 @@ export class TwaManifest {
+   * @param {URL} webManifestUrl the URL where the webmanifest is available.
+   * @param {WebManifest} webManifest the Web Manifest, used as a base for the update of
+   *    the TWA Manifest.
+   * @param {TwaManifest} oldTwaManifest current Twa Manifest.
+   */
+  static merge(fieldsToIgnore: string[], webManifestUrl: URL
+      , webManifest: WebManifestJson, oldTwaManifest: TwaManifest): void {
+    let shortcuts: ShortcutInfo[] = [];
+    let name;
+    let launcherName;
+    let display;
+    let themeColor;
+    let backgroundColor;
+    let fullStartUrl: URL;
+    let startUrl;
+    let icon: WebManifestIcon | null = null;
+    let maskableIcon: WebManifestIcon | null = null;
+    let monochromeIcon: WebManifestIcon | null = null;
+
+    const packageId = oldTwaManifest.packageId;
+    const host = oldTwaManifest.host;
+    const navigationColor = oldTwaManifest.navigationColor.hex();
+    const navigationColorDark = oldTwaManifest.navigationColorDark.hex();
+    const navigationDividerColor = oldTwaManifest.navigationDividerColor.hex();
+    const navigationDividerColorDark = oldTwaManifest.navigationDividerColorDark.hex();
+    const appVersionName = oldTwaManifest.appVersionName;
+    const signingKey = oldTwaManifest.signingKey;
+    const splashScreenFadeOutDuration = oldTwaManifest.splashScreenFadeOutDuration;
+    const enableNotifications = oldTwaManifest.enableNotifications;
+    const twaManifestShortcuts = shortcuts;
+    const twaManifestWebManifestUrl = oldTwaManifest.webManifestUrl!.toString();
+
+    function resolveIconUrl(icon: WebManifestIcon | null): string | undefined {
+      return icon ? new URL(icon.src, webManifestUrl).toString() : undefined;
+    }
+
+    if (!('shortcuts' in fieldsToIgnore)) {
+      for (let i = 0; i < (webManifest.shortcuts || []).length; i++) {
+        const s = webManifest.shortcuts![i];
+        try {
+          const shortcutInfo = ShortcutInfo.fromShortcutJson(webManifestUrl, s);
+          if (shortcutInfo != null) {
+            shortcuts.push(shortcutInfo);
+          }
+        } catch (err) {
+          TwaManifest.log.warn(`Skipping shortcut[${i}] for ${err.message}.`);
+        }
+        if (shortcuts.length === 4) {
+          break;
+        }
+      }
+    } else {
+      shortcuts = oldTwaManifest.shortcuts;
+    }
+
+    if (!('name' in fieldsToIgnore)) {
+      name = webManifest['name'] || webManifest['short_name'] || DEFAULT_APP_NAME;
+    } else {
+      name = oldTwaManifest.name;
+    }
+
+    if (!('launcherName' in fieldsToIgnore)) {
+      launcherName = webManifest['short_name'] ||
+        webManifest['name']?.substring(0, SHORT_NAME_MAX_SIZE) || DEFAULT_APP_NAME;
+    } else {
+      launcherName = oldTwaManifest.launcherName;
+    }
+
+    if (!('display' in fieldsToIgnore)) {
+      display = asDisplayMode(webManifest['display']!) || DEFAULT_DISPLAY_MODE;
+    } else {
+      display = oldTwaManifest.display;
+    }
```

Comment:
You're using this pattern a lot, I think we can simplify it:

```
function maybeUpdate(fieldName: string, fieldsToIgnore: string[],
    oldValue: T, newValue: T): T{
  if (fieldName in fieldsToIgnore) {
    return oldValue;
  }
  return newValue || oldValue;
}
```

(This is also making that change about ignoring default values I mentioned above.)

Then all of this code just becomes:

```
...
const display = maybeUpdate('display', fieldsToIgnore, oldTwaManifest.display
  asDisplayMode(webManifest['display']!));
const backgroundColor = maybeUpdate('backgroundColor', fieldsToIgnore,
  oldTwaManifest.backgroundColor.hex(), webManifest['background_color']);
...
```

Reply:
Nice! will do